apps/server/src/shared/domain/domainobject/board/board-composite.do.spec.ts

```diff
@@ -19,9 +25,9 @@ const buildBoardObject = () =>
 describe(`${BoardComposite.name}`, () => {
 	const setup = () => {
 		const parent = buildBoardObject();
-		parent.addChild(buildBoardObject() as AnyBoardDo);
-		parent.addChild(buildBoardObject() as AnyBoardDo);
-		parent.addChild(buildBoardObject() as AnyBoardDo);
+		parent.addChild(buildBoardObject() as unknown as AnyBoardDo);
```

Comment:
Is this a good test case if we work with as cast in file?

Reply:
The test is about the `BaseComposite` abstract base class. In order to not confuse the test with any of our concrete board DOs we are implementing a simple class `BoardObject` and a factory function `buildBoardObject` in the test file. Unfortunatetly `BoardObject` is not part of the union type `AnyBoardDo` we are using for the API of our domain objects. That's why the cast. We are planning to refactor `AnyBoardDo` to just `BoardComposite`. So the problem will eventually go away.

For the meantime I moved the cast to the factory function:
https://github.com/hpi-schul-cloud/schulcloud-server/pull/4072/commits/6dfbf9c7601ba058ba2a63636f54f3ac15ff5dde